libs/platform/src/lib/components/form/text-area/text-area.component.spec.ts

```diff
@@ -0,0 +1,336 @@
+
+        textareaElement.value = '';
+        await wait(fixture);
+
+        expect(host.form.get('basicTextarea').value).toBe('');
+    });
+
+    it('should submit the value', async () => {
+        const submitButton = host.submitButton.nativeElement;
+        submitButton.click();
+
+        await wait(fixture);
+
+        expect(host.result).toEqual({ basicTextarea: 'this is a random note' });
+    });
+});
+
+describe('Advanced Textarea', () => {
+    let fixture: ComponentFixture<BasicTextareaTestWrapperComponent>;
+    let host: BasicTextareaTestWrapperComponent;
+
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            imports: [ReactiveFormsModule, FdpFormGroupModule, PlatformTextAreaModule],
+            declarations: [BasicTextareaTestWrapperComponent]
+        }).compileComponents();
+    }));
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(BasicTextareaTestWrapperComponent);
+        host = fixture.componentInstance;
+
+        fixture.detectChanges();
+    });
+
+    async function wait(componentFixture: ComponentFixture<any>) {
+        componentFixture.detectChanges();
+        await componentFixture.whenStable();
+    }
+
+    it('should call validateLengthOnCustomSet', async () => {
+        const textareaElement = host.textareaComponent;
+        spyOn(textareaElement, 'validateLengthOnCustomSet');
+        textareaElement.maxLength = 5;
+        textareaElement.value = 'abcdefgg';
+
+        expect(textareaElement.validateLengthOnCustomSet).toHaveBeenCalled();
+    });
+
+    it('should change the counter message correctly', async () => {
+        const textareaElement = host.textareaComponent;
+        textareaElement.maxLength = 5;
+        textareaElement.value = 'abcdefgg';
+
+        await wait(fixture);
+
+        expect(host.form.get('basicTextarea').value).toBe('abcdefgg');
+
+        const textareaCounterElement = fixture.debugElement.query(By.css('.fd-textarea-counter'));
+        const result = textareaCounterElement.nativeElement.textContent.toString().trim();
+        expect(result).toBe('3 characters over the limit');
+    });
+
+    it('should change the counter message correctly when within limit', async () => {
+        const textareaElement = host.textareaComponent;
+        textareaElement.maxLength = 5;
+        textareaElement.value = 'abc';
+
+        await wait(fixture);
+
+        expect(host.form.get('basicTextarea').value).toBe('abc');
+
+        const textareaCounterElement = fixture.debugElement.query(By.css('.fd-textarea-counter'));
+        const result = textareaCounterElement.nativeElement.textContent.toString().trim();
+        expect(result).toBe('2 characters remaining');
+    });
+
+    // clipboard actions have stricter security and only work on secure https urls.
+    // https://developers.google.com/web/updates/2018/03/clipboardapi
+    // Therefore, ignoring this test.
+    xit('should handle paste interaction', async () => {
```

Comment:
@kavya-b if you don't want to have this test then can you just remove it?

Reply:
removed